cli/fs.rs

```diff
@@ -10,12 +10,40 @@ use rand;
 use rand::Rng;
 use walkdir::WalkDir;
 
+#[cfg(unix)]
+use nix::sys::stat::{mode_t, umask as unix_umask, Mode};
+
 #[cfg(unix)]
 use std::os::unix::fs::{DirBuilderExt, PermissionsExt};
 
 #[cfg(unix)]
 use nix::unistd::{chown as unix_chown, Gid, Uid};
 
+#[cfg(unix)]
+pub fn umask(mask: Option<u32>) -> Result<u32, ErrBox> {
+  let previous: Mode = unix_umask(Mode::from_bits_truncate(
+    mask.unwrap_or(0o777) as mode_t & 0o777,
+  ));
+  if mask.is_none() {
+    unix_umask(previous);
+  }
+  Ok(previous.bits() as u32 & 0o777)
+}
+
+#[cfg(not(unix))]
+pub fn umask(mask: Option<u32>) -> Result<u32, ErrBox> {
+  match mask {
+    Some(_) => {
+      let e = std::io::Error::new(
+        std::io::ErrorKind::Other,
+        "Not implemented".to_string(),
+      );
+      Err(ErrBox::from(e))
+    }
+    None => Ok(0),
+  }
+}
```

Comment:
Can you move this code into the op_umask implementation itself? I don't think it's useful to have these separate methods in cli/fs.rs

Reply:
I was going to do it later. As emerged in an earlier exchange, I wasn't clear on the intended division of labor between cli/fs.rs and cli/ops/fs.rs. So I implemented things in a way that I now realize you'd prefer to be folded into cli/ops/fs.rs. What I was planning to do was get this queue of PRs out for your review --- it's already complex keeping up with the changing master branch, waiting to push later PRs until earlier ones that block them have been merged, and so on. And then afterward, move all the stuff that doesn't need to be in cli/fs.rs into cli/ops/fs.rs. But if you prefer me to move stuff into cli/ops/fs.rs along the way, I'll do so.